Sensors/SMCProcessor/SMCProcessor.cpp

```diff
@@ -56,9 +61,85 @@ void SMCProcessor::readRapl() {
 	}
 }
 
-void SMCProcessor::updateCounters() {
-	uint32_t cpu = cpu_number();
+IOReturn SMCProcessor::bindCurrentThreadToCpu(uint32_t cpu)
+{
+	// Obtain power management callbacks
+	pmCallBacks_t callbacks {};
+	pmKextRegister(PM_DISPATCH_VERSION, nullptr, &callbacks);
+	
+	if (!callbacks.LCPUtoProcessor) {
+		SYSLOG("scpu", "failed to obtain LCPUtoProcessor");
+		return KERN_FAILURE;
+	}
+	
+	if (!callbacks.ThreadBind) {
+		SYSLOG("scpu", "failed to obtain ThreadBind");
+		return KERN_FAILURE;
+	}
+	
+	auto preemptionLock = IOSimpleLockAlloc();
+	if (!preemptionLock) {
+		SYSLOG("scpu", "Preemption lock cannot be allocated");
+		return KERN_FAILURE;
+	}
+	
+	if (!IOSimpleLockTryLock(preemptionLock)) {
```

Comment:
Why is that `Try` lock instead of just lock?

Reply:
IOSimpleLockTryLock is used in sample code by @07151129. But you are right, I think we definitely can use IOSimpleLock.